Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.h

```diff
@@ -62,10 +62,12 @@ namespace AZ::DocumentPropertyEditor
         //! property editor instances has altered its value.
         void NotifyPropertyChanged(const PropertyChangeInfo& changeInfo);
 
-        //! Invoked when the adapter builder builds a new row.
+        //! Creates a node for displaying label information.
+        //! Subclass adapter can override the default behavior.
         //! @param adapterBuilder The adapter builder currently being used by this adapter.
-        //! @param serializedpath The serialized path fetched from AZ::Reflection::DescriptorAttributes.
-        virtual void OnBeginRow(AdapterBuilder* adapterBuilder, AZStd::string_view serializedPath);
+        //! @param labelText The text string to be displayed in label.
+        //! @param serializedPath The serialized path fetched from AZ::Reflection::DescriptorAttributes.
+        virtual void CreateLabel(AdapterBuilder* adapterBuilder, AZStd::string_view labelText, AZStd::string_view serializedPath);
```

Comment:
I like this approach, personally. I hope we can eventually add more virtual node/property construction methods in this class (biggest roadblock to that is the potential to break stuff though). Makes it easier for derived adapters to override/extend the inherited behavior and better organization here might make the code easier to maintain.

I'm not sure how many adapters will override the ReflectionAdapter, but I think it's a good pattern for people to refer to when making their own adapters.